File: freight-triage-polyglot-ledger-recovery/environment/native/src/norm/pad_left_eight.cpp
#include "freight/normalize.h"

#include <algorithm>

namespace freight {

// Normalizer: pad left eight.
std::string norm_pad_left_eight(const std::string& text) {
  if (text.size() >= 8) {
    return text;
  }
  return std::string(8 - text.size(), '0') + text;
}

}  // namespace freight

File: freight-triage-polyglot-ledger-recovery/environment/intake/src/com/freight/norm/PadLeftEight.java
package com.freight.norm;

/** Normalizer: pad left eight. */
public final class PadLeftEight implements Normalizer {

    @Override
    public String name() {
        return "pad_left_eight";
    }

    @Override
    public String apply(String text) {
        if (text.length() >= 8) {
            return text;
        }
        StringBuilder out = new StringBuilder();
        for (int i = text.length(); i < 8; i++) {
            out.append('0');
        }
        return out.append(text).toString();
    }
}
